benchmarks/bouncy_ss.jl

```diff
@@ -0,0 +1,177 @@
+using Tilde 
+using ProgressMeter
+using LinearAlgebra
+using Revise
+using Optim
+
+using ZigZagBoomerang
+const ZZB = ZigZagBoomerang
+using LinearAlgebra
+using DelimitedFiles
+using Random
+using ForwardDiff
+using ForwardDiff: Dual
+using ZigZagBoomerang.PDMats
+using MappedArrays
+
+N = 100000 # no. observations
+C = 400 # no. covariates
+Random.seed!(1)
+A = rand(1:C, N) # covariates selected at random
+
+# The model we are interested in
+full_model = @model N1, C, A, y begin
+    α ~ Normal()
+    cc ~ Normal()^C
+    for i in 1:N1
+        v = α + cc[A[i]]
+        y[i] ~ Bernoulli(logitp = v)
+    end
+end
+
+# This is an model which gives unbiased gradient estimates of the full model
+# by subsampling using MeasureTheory's `PowerWeightedMeasure`
+ss_model = @model N1, C, A, K, seed, y begin
+    # same prior
+    α ~ Normal()
+    cc ~ Normal()^C
+    # but random subset of observations
+    sampler = Random.SamplerRangeNDL(1:N1)
+    rng = ZZB.Rng(seed)
+    for _ in 1:K
+        i = rand(rng, sampler) # each index expected to be sampled (K/N)*n times in n calls
```

Comment:
```suggestion
    for i in lazyrand(ZZB.Rng(seed), 1:N1, K)
```
with a lazy version of `rand(rng, range, n)`

Reply:
```
struct LazyRand{T,R}
    sampler::T
    n::Int
    rng::R
end

Base.iterate(iter::LazyRand) = iter.n > 0 ?  (rand(iter.rng, iter.sampler),iter.n-1) : nothing
Base.iterate(iter::LazyRand, n) = n > 0 ? (rand(iter.rng, iter.sampler), n-1) : nothing
lazyrand(rng, r::UnitRange, n) = LazyRand(Random.SamplerRangeNDL(r), n, rng)
lazyrand(r::UnitRange, n) = LazyRand(Random.SamplerRangeNDL(r), n, Random.GLOBAL_RNG)
Base.length(iter::LazyRand) = iter.n
```